pkg/destroy/azure/azure.go

```diff
@@ -419,17 +423,63 @@ func deleteAzureStackPublicRecords(ctx context.Context, o *ClusterUninstaller) e
 	return utilerrors.NewAggregate(errs)
 }
 
-func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, privateDNSClient privatedns.PrivateZonesClient, privateRecordsClient privatedns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
+func deleteRecordsFromBaseDomain(ctx context.Context, o *ClusterUninstaller) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	if o.BaseDomainResourceGroupName == "" || o.ZoneName == "" || o.APIURL == "" {
+		o.Logger.Debugf("could not find values in the metadata to get record set")
+		return nil
+	}
+
+	var errs []error
+	tag := fmt.Sprintf("kubernetes.io_cluster.%s", o.InfraID)
+	result, err := o.recordsClient.Get(ctx, o.BaseDomainResourceGroupName, o.ZoneName, o.APIURL, dns.CNAME)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to get base domain dns zone: %w", err))
+		if isAuthError(err) {
+			return err
+		}
+	}
+
+	if value, ok := result.Metadata[tag]; ok {
+		if *value == "owned" {
+			deleteResult, err := o.recordsClient.Delete(ctx, o.BaseDomainResourceGroupName, o.ZoneName, o.APIURL, dns.CNAME, "")
+			if err != nil {
+				if deleteResult.IsHTTPStatus(http.StatusNotFound) {
+					o.Logger.Debug("already deleted")
+					return utilerrors.NewAggregate(errs)
+				}
+				errs = append(errs, fmt.Errorf("failed to delete base domain dns zone: %w", err))
+				if isAuthError(err) {
+					return err
+				}
+			} else {
+				o.Logger.WithField("record", to.String(&o.APIURL)).Info("deleted")
+			}
+		}
+	} else {
+		o.Logger.Debugf("metadata mismatch")
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func deletePublicRecords(ctx context.Context, o *ClusterUninstaller) error {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
 	defer cancel()
 
 	// collect records from private zones in rgName
 	var errs []error
 
-	zonesPage, err := dnsClient.ListByResourceGroup(ctx, rgName, to.Int32Ptr(100))
+	zonesPage, err := o.zonesClient.ListByResourceGroup(ctx, o.ResourceGroupName, to.Int32Ptr(100))
 	if err != nil {
 		if zonesPage.Response().IsHTTPStatus(http.StatusNotFound) {
-			logger.Debug("already deleted")
+			o.Logger.Debug("already deleted")
```

Comment:
Technically we don't know it was deleted. Maybe something to convey: private zone was not found, checking public records...